components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -122,6 +135,43 @@ private String resolveUserIdFromUsername(int tenantId, String userStoreDomain, S
         }
     }
 
+    /**
+     * Retrieves the username of the given userId.
+     *
+     * @param tenantId Id of the tenant domain of the user.
+     * @param userId Id of the user.
+     * @return username.
+     * @throws UserSessionException
+     */
+    private String getUsernameFromUserId(int tenantId, String userId) throws
+            UserSessionException {
+
+        try {
+            UserStoreManager userStoreManager = FrameworkServiceComponent.getRealmService()
+                    .getTenantUserRealm(tenantId).getUserStoreManager();
+            try {
+                if (userStoreManager instanceof AbstractUserStoreManager) {
+                    return ((AbstractUserStoreManager) userStoreManager).getUserNameFromUserID(userId);
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Provided user store manager for the tenantId: " + tenantId + ", is not an instance " +
+                            "of the AbstractUserStore manager.");
+                }
+                throw new UserSessionException("Unable to get the username for the userId: " + userId + ".");
+            } catch (org.wso2.carbon.user.core.UserStoreException e) {
```

Comment:
Check and verify two different exceptions with `UserStoreException`. Remove the redundent

Reply:
These are existing methods and are returning these two types of `UserStoreException`